marimo/_utils/narwhals_utils.py

```diff
@@ -41,14 +41,14 @@ def assert_narwhals_series(series: nw.Series) -> None:
         raise ValueError(f"Unsupported series type. Got {type(series)}")
 
 
-def can_narwhalify(obj: Any) -> TypeGuard[IntoFrame]:
+def can_narwhalify(obj: Any, eager_only: bool = False) -> TypeGuard[IntoFrame]:
     """
     Check if the given object can be narwhalified.
     """
     if obj is None:
         return False
     try:
-        nw.from_native(obj, strict=True)
+        nw.from_native(obj, strict=True, eager_only=eager_only)  # type: ignore[call-overload]
```

Comment:
I was curious about the performance of this method, and saw it uses the same trick as we do -- getting modules from `sys.modules` instead of importing them ✅

Intentional to not allow series?

Reply:
series is true by default. i can make that explicit (in case defaults change and readability)

Reply:
i think i saw a narwhals PR too to add this utility without the try/catch